### ISSUE TITLE
Routes are not found when using Microsoft.AspNetCore.All

### ISSUE THREAD

**Issue by sebastienros** [aspnet/Mvc#7605]
In the benchmarks app    

```
    <PackageReference Include="Microsoft.AspNetCore.All" Version="$(BenchmarksAspNetCoreVersion)" />
```

I can't reach mvc endpoints, but when I use
```
    <PackageReference Include="Microsoft.AspNetCore.App" Version="$(BenchmarksAspNetCoreVersion)" />
    <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv" Version="$(BenchmarksAspNetCoreVersion)" />
```
Then mvc is enabled, but I get a 404 and a debug message stating no route matched.

This application uses attribute routing, so I assume the controller is not found in this case.
All other middleware work as expected.
I tried with the latest aspnet core version and also a one week old one.

Note: I have not yet tried the templates to repro ... doing it now

**Comment by Eilon** [comment]
@mkArtakMSFT - can we have someone take a look first thing tomorrow w/ Preview2 to see if this is something we would reset on?

**Comment by rynowak** [comment]
For an extremely short period of time we had a build of the SDK that would produce a messed up deps file in some really specific circumstances, I'm wondering if it's possible that this is that. 

@pranavkm would know remember what the workaround for that was (an msbuild thing) would help narrow this down

**Comment by pranavkm** [comment]
We fixed this in the most recent SDK. But the fix was to set `<PreserveCompilationContext>true</PreserveCompilationContext>` in your app if you're blocked on updating the Sdk.

**Comment by sebastienros** [comment]
We already tried it with @JunTaoLuo  and it didn't solve it

**Comment by pranavkm** [comment]
Yeah, this is broken. We have to explicitly list `Microsoft.AspNetCore.All` here - https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/ApplicationAssembliesProvider.cs#L17 - the deps file does not say that `Microsoft.AspNetCore.All` depends on `Microsoft.AspNetCore.Mvc` which breaks our search.

**Comment by pranavkm** [comment]
Fixed in 5e019bd707497b0d28a7387c7ed66447a1907d59
